tracer/src/Datadog.Trace.Tools.Runner/CheckIisCommand.cs

```diff
@@ -0,0 +1,147 @@
+// <copyright file="CheckIisCommand.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tools.Runner.Checks;
+using Microsoft.Web.Administration;
+using Spectre.Console;
+using Spectre.Console.Cli;
+
+using static Datadog.Trace.Tools.Runner.Checks.Resources;
+
+namespace Datadog.Trace.Tools.Runner
+{
+    internal class CheckIisCommand : AsyncCommand<CheckIisSettings>
+    {
+        public override Task<int> ExecuteAsync(CommandContext context, CheckIisSettings settings)
+        {
+            return ExecuteAsync(settings, null, null);
+        }
+
+        public override ValidationResult Validate(CommandContext context, CheckIisSettings settings)
+        {
+            var result = base.Validate(context, settings);
+
+            if (result.Successful)
+            {
+                // Perform additional validation
+                if (settings.SiteName.Count(c => c == '/') > 1)
+                {
+                    return ValidationResult.Error($"IIS site names can't have multiple / in their name: {settings.SiteName}");
+                }
+            }
+
+            return result;
+        }
+
+        internal static async Task<int> ExecuteAsync(CheckIisSettings settings, string applicationHostConfigurationPath, int? pid)
+        {
+            var values = settings.SiteName.Split('/');
+
+            var siteName = values[0];
+            var applicationName = values.Length > 1 ? $"/{values[1]}" : "/";
+
+            AnsiConsole.WriteLine(FetchingApplication(siteName, applicationName));
+
+            var serverManager = new ServerManager(readOnly: true, applicationHostConfigurationPath);
+
+            var site = serverManager.Sites[siteName];
+
+            if (site == null)
+            {
+                Utils.WriteError(CouldNotFindSite(siteName, serverManager.Sites.Select(s => s.Name)));
+
+                return 1;
+            }
+
+            var application = site.Applications[applicationName];
+
+            if (application == null)
+            {
+                Utils.WriteError(CouldNotFindApplication(siteName, applicationName, site.Applications.Select(a => a.Path)));
+
+                return 1;
+            }
+
+            var pool = serverManager.ApplicationPools[application.ApplicationPoolName];
+
+            // The WorkerProcess part of ServerManager doesn't seem to be compatible with IISExpress
+            // so we skip this bit when launched from the tests
+            if (pid == null)
```

Comment:
How are we detecting that? AFAICT, we pass in a pid into the tests too?

Reply:
Well yes, we **only** pass a PID from the tests. I don't plan on supporting IIS Express in the wild.